tests/lib/rules/newline-per-chained-call.js

```diff
@@ -116,13 +153,19 @@ ruleTester.run("newline-per-chained-call", rule, {
             "    'method3' :",
             "    'method4']()"
         ].join("\n"),
+        output: [
+            "anObject.method1().method2()\n['method' + n]()\n[aCondition ?",
+            "    'method3' :",
+            "    'method4']()"
+        ].join("\n"),
         errors: [{
             message: "Expected line break before `['method' + n]`."
         }, {
             message: "Expected line break before `[aCondition ?`."
         }]
     }, {
         code: "foo.bar()['foo' + \u2029 + 'bar']()",
+        output: "foo.bar()\n['foo' + \u2029 + 'bar']()",
```

Comment:
Can you add a test where there is a comment between the object and the property?

```js
foo . /* comment */ bar();
```

(I'm not really sure what output I would expect for this case since the code is sort of strange, so pretty much anything would be fine as long as it doesn't delete the comment and doesn't produce invalid syntax.)

Reply:
You were right about the parenthesized properties/object, so I updated the `fix()` with your recommendation, and everything is passing accordingly. Regarding this comment, I would expect this to break before the `.` like:
```js
foo
. /* comment */ bar();
```

In my opinion the comment refers to the property, so it should break before the comment. Without further changes everything to the current implementation, it is working as expected. I added another test with the comment before the dot, and it breaks the line correctly as well:

```js
foo /* comment */
.bar();
```

Reply:
For what it's worth, in any case where we're unsure of the expected fix behavior, we can always choose not to fix. I think these cases where block comments make it difficult to infer what the user would really want would be a great place not to fix. Just my two cents.

Reply:
If we opt to not fix it I think we need to change the rule in order to not report for block comments. The expected behaviour of the autofixer is to perform fixes, and if it leaves unfixed code after running it would break pre commit hooks, etc. There are other rules such as `array-element-newline` that fix it anyway like:

```js
/* eslint array-element-newline: error */
const foo = [1,2 /* comment */,3];

// autofixes to:
const foo = [1,
2 /* comment */,
3];
```

Reply:
I don't feel strongly about this particular case, but there is precedent for autofixers not fixing all problems. In general we don't make any guarantees about whether a given autofix will be applied, and some rules aren't autofixable at all. For example, `brace-style` doesn't remove the linebreak before the `{` in cases like this:

```js
if (foo) // comment
{
    bar();
}
```

And `eqeqeq` can only autofix a small number of cases where a `typeof` expression or a literal is used.

Reply:
But in the case of `brace-style` it is needed to get the final brace position since it would affect the final output. I think that in this case in this particular case it is ok to fix it anyway since it just fixes the rule as it is, but it is up to you.

Reply:
Seems reasonable to me.